src/dom/Boolean.ts

```diff
@@ -75,13 +75,12 @@ export class Boolean extends Value(
   ): boolean {
     let isSupportedType: boolean = false;
     let valueToCompare: any = null;
-    if (options.onlyCompareIon) {
-      // `compareOnlyIon` requires that the provided value be an ion.dom.Boolean instance.
-      if (other instanceof Boolean) {
-        isSupportedType = true;
-        valueToCompare = other.booleanValue();
-      }
-    } else {
+    // `compareOnlyIon` requires that the provided value be an ion.dom.Boolean instance.
```

Comment:
Now that we're not checking `if (options.onlyCompareIon)` here, I think this comment may be confusing. (Same thought in the other similarly refactored areas.)